core/modules/layout/layout.module

```diff
@@ -1214,83 +1214,54 @@ function layout_get_layout_by_path($path = NULL, $router_item = NULL) {
     $router_item = menu_get_item($path);
   }
 
-  $cache = &backdrop_static(__FUNCTION__);
-
-  if (empty($cache[$router_item['path']])) {
-    // If the path is a node preview path, check if a custom layout exists for
-    // nodes and use that instead.
-    // @ todo: This adds very specific code for node previews which too closely
-    // couples Node and Layout modules. This needs to be reverted in a future
-    // commit.
-    $preview_node = NULL;
-    $node_types = node_type_get_types();
-    foreach ($node_types as $type) {
-      $preview_path = 'node/preview/' . $type->type . '/%';
-      if ($router_item['path'] == $preview_path) {
-        $tempstore_id = $router_item['map'][3];
-        $preview_node = node_get_node_tempstore($tempstore_id);
-        break;
-      }
-    }
-    if ($preview_node) {
-      $layouts = layout_load_multiple_by_path('node/%', TRUE);
-    }
-    else {
-      $layouts = layout_load_multiple_by_path($router_item['path'], TRUE);
-    }
-    $selected_layout = NULL;
-    foreach ($layouts as $layout) {
-      // Contexts must be set before the layout's access may be checked.
-      $contexts = $layout->getContexts();
-      foreach ($contexts as $context) {
-        if (isset($context->position)) {
-
-          // If on a node preview page, use the preview from tempstore.
-          // @ todo: This is also specific code for node previews which needs
-          // to be reverted in a future commit.
-          if ($preview_node) {
-            $context->setData($preview_node);
+  // Use the static cache, but based on the normal path (not system path).
+  $selected_layout = &backdrop_static(__FUNCTION__ . '/' . $router_item['href']);
```

Comment:
This would complicate resetting the static variable. To call `backdrop_static_reset('layout_get_layout_by_path/' . $href)` would be cumbersome for one, and you'd need to know what is `$href` while calling.

Reply:
True, good point. I can go back to the previous approach that uses `&backdrop_static(__FUNCTION__)`, but still use `'href'` as the key.